tarantool_test.go

```diff
@@ -2830,6 +2831,115 @@ func TestStream_DoWithClosedConn(t *testing.T) {
 	}
 }
 
+func TestConnectionIdInfo(t *testing.T) {
+	conn := test_helpers.ConnectWithValidation(t, server, opts)
+	defer conn.Close()
+
+	isLess210, err := test_helpers.IsTarantoolVersionLess(2, 10, 0)
+	if err != nil {
+		t.Fatalf("Unexpected error has been caught: %s", err.Error())
+	}
+	// First Tarantool protocol version (1) was introduced between
+	// 2.10.0-beta1 and 2.10.0-beta2. Versions 2 and 3 were also
+	// introduced between 2.10.0-beta1 and 2.10.0-beta2. Version 4
+	// was introduced between 2.10.0-beta2 and 2.10.0-rc1 and reverted
+	// back to version 3 in the same version interval.
+	// So each release Tarantool >= 2.10 have protocol version >= 3.
+	// (Tarantool 2.10.4 still has version 3.)
+	tarantool210ProtocolVersion := ProtocolVersion(3)
+
+	require.Equal(t, conn.ClientProtocolVersion(), ProtocolVersion(3))
+
+	require.ElementsMatch(t,
+		conn.ClientFeatures(),
+		[]Feature{StreamsFeature, TransactionsFeature})
+
+	if isLess210 {
+		require.Equal(t, conn.ServerProtocolVersion(), NoProtocolVersion)
+		require.ElementsMatch(t, conn.ServerFeatures(), []Feature{})
+	} else {
+		require.GreaterOrEqual(t,
+			conn.ServerProtocolVersion(),
+			tarantool210ProtocolVersion)
+
+		require.Subset(t,
+			conn.ServerFeatures(),
+			[]Feature{
+				StreamsFeature,
+				TransactionsFeature,
+				ErrorExtensionFeature,
+				WatchersFeature})
+	}
```

Comment:
It would be easier to split the test into two.

Reply:
Changed